docs(core): add Ivy note to entryComponents page (#35479)
angular/angular: aio/content/guide/entry-components.md

# Entry components

<div class="alert is-helpful">

**Note:** Entry Components have been deprecated and are no longer used with [Ivy rendering engine](https://angular.io/guide/ivy). 

> Learn more about [entryComponents deprecation](https://angular.io/guide/deprecations#entrycomponents-and-analyze_for_entry_components-no-longer-required) in the [Deprecated APIs and features](https://angular.io/guide/deprecations) page.

Reviewer: ```suggestion
Learn more about [entryComponents deprecation](https://angular.io/guide/deprecations#entrycomponents-and-analyze_for_entry_components-no-longer-required) on the [Deprecated APIs and features](https://angular.io/guide/deprecations) page.
```

> **Note:** Entry Components have been deprecated and are no longer used with [Ivy rendering engine](https://angular.io/guide/ivy). 
> Learn more about [entryComponents deprecation](https://angular.io/guide/deprecations#entrycomponents-and-analyze_for_entry_components-no-longer-required) on the [Deprecated APIs and features](https://angular.io/guide/deprecations) page.

Reviewer 2: ```suggestion
Entry components have been deprecated with the [Ivy rendering engine](https://angular.io/guide/ivy). 
For more information, see [entryComponents deprecation](https://angular.io/guide/deprecations#entrycomponents-and-analyze_for_entry_components-no-longer-required) in the [Deprecated APIs and features](https://angular.io/guide/deprecations).
```

you won't have to explicitly set any entry components because Angular adds components listed in `@NgModule.bootstrap` and those in route definitions to entry components automatically. Though these two mechanisms account for most entry components, if your app happens to bootstrap or dynamically load a component by type imperatively,
you must add it to `entryComponents` explicitly.

<div class="alert is-helpful">

> **Note:** Since 9.0.0. With Ivy, `entryComponents` property is no longer necessary.

Reviewer: Thanks for creating this PR, adding a message to the `entryComponents` section of the guide makes sense.

I'd propose a couple updates:

- Move it at the very top of the section, so that developers who use Ivy can stop reading this section (since this information is not relevant for them).
- Add a reference to this page where more information can be found: https://angular.io/guide/deprecations#entrycomponents-and-analyze_for_entry_components-no-longer-required
- I'd also drop the version and just mention that it's applicable to apps that use Ivy rendering engine (and reference this page: https://angular.io/guide/ivy).

Thank you.

Reviewer 3: []()
